pypeit/biasframe.py

```diff
@@ -71,16 +73,25 @@ def build_image(self, overwrite=False, trim=True):
 
         Args:
             overwrite: (:obj: `bool`, optional):
-                Regenerate the stack image
+                Regenerate the combined image
             trim (:obj:`bool`, optional):
                 If True, trim the image
 
         Returns:
             `numpy.ndarray`_: Combined, processed image.
         """
-        if self.par['useframe'] == 'none':
+        # Nothing?
+        if self.par['useframe'].lower() == 'none':
+            msgs.info("Bias image subtraction not activated.")
+            return None
+        if self.nfiles == 0:
+            msgs.info("No bias frames provided.  No bias image will be generated or used")
             return None
-        return self.process(bias_subtract=None, trim=trim, overwrite=overwrite)
+        # Load
```

Comment:
The way this is coded up it is going to be a massive memory hog. You are storing all the copies of the raw images in memory, and then later on you are going image by image and processing them. I don't understand why you would not simply process them image by image. We never do anything with these raw unprocessed frames so I see no reason to store them in memory first, and then loop over them and process them one by one. I think the interface should be that the processing all takes place in the single image interface, i.e. processimage, and that combined image only operates on already processed things.